azurerm/internal/services/containers/data_source_kubernetes_cluster.go

```diff
@@ -212,8 +212,18 @@ func dataSourceArmKubernetesCluster() *schema.Resource {
 			},
 
 			"private_link_enabled": {
-				Type:     schema.TypeBool,
-				Computed: true,
+				Type:          schema.TypeBool,
+				Computed:      true,
+				Optional:      true,
+				ConflictsWith: []string{"enable_private_cluster"},
+				Deprecated:    "Deprecated in favor of `enable_private_cluster`", // TODO -- remove this in next major version
+			},
+
+			"enable_private_cluster": {
```

Comment:
could we be more consistent with the rest of the provider and name this
```suggestion
			"private_cluster_enabled": {
```

Reply:
Thanks! Fixed as suggested